pallets/pallet-deposits/src/deposit_change_handler.rs

```diff
@@ -0,0 +1,39 @@
+use frame_system::pallet_prelude::OriginFor;
+use sp_runtime::DispatchResult;
+
+use crate::{Config, DepositIdOf, AccountIdOf, BalanceOf};
+
+/// DepositChangeHandler is meant to be implemented on the runtime level to allow it to react to the event when a deposit changed
+pub trait DepositChangeHandler<T: Config> {
```

Comment:
This trait should probably be generic over `Deposit`, `AccountId` and `Balance` rather than constraining the type. It makes it easier to mock and test.